File: Index_Electric_server/src/main/java/com/index_electric_server/reading/dto/ReadingPullItem.java
package com.index_electric_server.reading.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class ReadingPullItem {

    private String meterId;
    private String month;

    private Integer indexPrevMonth;
    private Integer indexLastMonth;
    private Integer indexConsumption;

    private LocalDateTime createdAt;

    public ReadingPullItem() {}

    public String getMeterId() {
        return meterId;
    }

    public void setMeterId(String meterId) {
        this.meterId = meterId;
    }

    public String getMonth() {
        return month;
    }

    public void setMonth(String month) {
        this.month = month;
    }

    public Integer getIndexPrevMonth() {
        return indexPrevMonth;
    }

    public void setIndexPrevMonth(Integer indexPrevMonth) {
        this.indexPrevMonth = indexPrevMonth;
    }

    public Integer getIndexLastMonth() {
        return indexLastMonth;
    }

    public void setIndexLastMonth(Integer indexLastMonth) {
        this.indexLastMonth = indexLastMonth;
    }

    public Integer getIndexConsumption() {
        return indexConsumption;
    }

    public void setIndexConsumption(Integer indexConsumption) {
        this.indexConsumption = indexConsumption;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime capturedAt) {
        this.createdAt = capturedAt;
    }
}

File: electric_index/lib/models/reading_pull_item.dart
class ReadingPullItem {
  final String meterId;
  final String month;
  final int indexPrevMonth;
  final int indexLastMonth;
  final int indexConsumption;
  final String? createdAt;

  ReadingPullItem({
    required this.meterId,
    required this.month,
    required this.indexPrevMonth,
    required this.indexLastMonth,
    required this.indexConsumption,
    this.createdAt,
  });

  factory ReadingPullItem.fromJson(Map<String, dynamic> j) {
    return ReadingPullItem(
      meterId: j['meterId']?.toString() ?? '',
      month: j['month']?.toString() ?? '',
      indexPrevMonth: (j['indexPrevMonth'] as num?)?.toInt() ?? 0,
      indexLastMonth: (j['indexLastMonth'] as num?)?.toInt() ?? 0,
      indexConsumption: (j['indexConsumption'] as num?)?.toInt() ?? 0,
      createdAt: j['createdAt']?.toString(),
    );
  }
}
